### ISSUE TITLE
kdash doesn't respect HTTPS_PROXY environment variable

### ISSUE THREAD

**Issue by b3ndill** [kdash-rs/kdash#380]
**Describe the bug**
I have a cluster on a private network that I need to access via a bastion host. Under normal circumstances I can use `kubectl` by including a `proxy-url` parameter in the Kubernetes context and setting the `HTTPS_PROXY` environment variable.
With `kdash` I get a timeout error: `Failed to get namespaces. HyperError(hyper::Error(Connect, Custom { kind: TimedOut, error: Elapsed(()) }))`

**To Reproduce**
1. Set a `proxy-url` parameter for your private Kubernetes cluster
2. Set up a port-forward connection via a jump host using SSH
3. Export the forwarded port to `HTTPS_PROXY` variable.
4. Launch `kdash`

**Expected behavior**
`kdash` should utilize the `HTTPS_PROXY` environment variable, if it is set, when trying to connect to the Kubernetes cluster.

**Desktop:**
 - OS and OS version: macOS 13.5
 - Terminal name and version: iTerm2 3.4.20
 - Shell name and version zsh
 - Kubectl Client Version: v1.26.7
 - Kubectl Server Version: v1.26.5
 - KDash version 0.4.2

**Comment by deepu105** [comment]
might be related to https://github.com/kube-rs/kube/issues/1013 need to dig in further

**Comment by b3ndill** [comment]
Just a heads-up, it seems like SOCKS5 support has been merged in https://github.com/kube-rs/kube/pull/1311. It should be available (behind a feature flag) in [release 0.87.1](https://github.com/kube-rs/kube/releases/tag/0.87.1).

**Comment by deepu105** [comment]
Yes, the kuber version needs to be updated as soon as other dependencies
(kubectl-view-allocations) also update to that version

Thanks & Regards,
Deepu

**Comment by deepu105** [comment]
I have enabled sock5 feature in version 0.6

**Comment by deepu105** [comment]
can you test and confirm  plz

**Comment by b3ndill** [comment]
> can you test and confirm plz

I've updated kdash but I'm not getting any cluster information. It sees the correct context but isn't capable of connecting to the cluster. However, kubectl works fine from the same terminal session

**Comment by deepu105** [comment]
Can you pass me your kube/config after removing private/sensitive info (I'm mainly interested in how the cluster authentication is structured)

Also what do you see if you run `kdash --debug`
